test/e2e/specs/specs-playwright/wp-theme__details-preview-activate.js

```diff
@@ -33,6 +34,16 @@ describe( DataHelper.createSuiteTitle( 'Theme: Preview and Activate' ), () => {
 		await sidebarComponent.gotoMenu( { item: 'Appearance', subitem: 'Themes' } );
 	} );
 
+	it( 'Select test site', async function () {
+		const siteSelectComponent = new SiteSelectComponent( page );
+
+		if ( await siteSelectComponent.isSiteSelectorVisible() ) {
+			await siteSelectComponent.selectSite(
+				DataHelper.getAccountSiteURL( user, { protocol: false } )
+			);
+		}
+	} );
```

Comment:
It seems to me that this step has a scenario where the site selector is not visible so it kinda gets skipped. If it's true, I think we should change the title to something like `Select test site if site selector present`. If it's not true, I think we should make the test throw if the site selector is not present as it would then be expected to be present. If the latter is our case, I think the `isSiteSelectorVisible` check should be called from within the `selectSite` method as a validator.

Reply:
Yes, this step is an optional one that should only run if the Select Site page intercepts the navigation to Appearance > Themes. 

There is an alternative way to do this however: [conditionally run step in Jest](https://stackoverflow.com/questions/58264344/conditionally-run-tests-in-jest).

So it could look something like this:

```
const itif = (condition) => condition ? it : it.skip;

// in the previous step, navigate to Appearance > Theme in this case
it (.....) {
    siteSelectorShown = await siteSelectComponent.isSiteSelectorVisible();
}

itif( 'Select test site') {
    // optional calls here
}

```

Reply:
The benefit of using the approach above is that, if the step is indeed skipped due to it being unnecessary, the step in question will not appear in the resulting logs.

Reply:
But I think it might be okay for now to change the name to indicate it is an optional step.